src/hevm/src/EVM.hs

```diff
@@ -2333,20 +2327,21 @@ costOfPrecompile (FeeSchedule {..}) precompileAddr input =
     -- MODEXP
     0x5 -> num $ (f (num (max lenm lenb)) * num (max lene' 1)) `div` (num g_quaddivisor)
       where (lenb, lene, lenm) = parseModexpLength input
-            lene' = if lene <= 32 && ez then 0
-                    else if lene <= 32 then num (log2 e')
-                    else if e' == 0 then 8 * (lene - 32)
-                    else num (log2 e') + 8 * (lene - 32)
+            lene'
+              | lene <= 32 && ez = 0
+              | lene <= 32 = num (log2 e')
+              | e' == 0 = 8 * (lene - 32)
+              | otherwise = num (log2 e') + 8 * (lene - 32)
 
             ez = isZero (96 + lenb) lene input
             e' = w256 $ word $ LS.toStrict $
                    lazySlice (96 + lenb) (min 32 lene) input
 
             f :: Integer -> Integer
-            f x = if x <= 64 then x * x
-                  else if x <= 1024
-                  then (x * x) `div` 4 + 96 * x - 3072
-                  else (x * x) `div` 16 + 480 * x - 199680
+            f x
+              | x <= 64 = x * x
+              | x <= 1024 = (x * x) `div` 4 + 96 * x - 3072
+              | otherwise = (x * x) `div` 16 + 480 * x - 199680
```

Comment:
```suggestion
            f x | x <= 64 = x * x
                | x <= 1024 = (x * x) `div` 4 + 96 * x - 3072
                | otherwise = (x * x) `div` 16 + 480 * x - 199680
```